src/subscript/eclcompress/eclcompress.py

```diff
@@ -312,19 +339,27 @@ def find_keyword_sets(
     """
     keywordsets = []
     kwstart = None
-    if eclkw_regexp is None:
-        eclkw_regexp = "[A-Z]{2,8}$"
-    if isinstance(eclkw_regexp, str):
+    if eclkw_regexp:
         eclkw_regexp = re.compile(eclkw_regexp)
 
     for lineidx, line in enumerate(filelines):
-        if (
-            re.match(eclkw_regexp, line) is not None
-            and line.strip() not in DENYLIST_KEYWORDS
-        ):
-            kwstart = lineidx
+        line = line.strip()
+        if not line:
             continue
-        if kwstart is not None and line.strip()[0:2] == "--":
+        if eclkw_regexp:
+            if re.match(eclkw_regexp, line) and line not in DENYLIST_KEYWORDS:
+                kwstart = lineidx
+                continue
+        else:
```

Comment:
@mferrera does it mean that if user specify the regular expression, it will ignore the ALLOWLIST_KEYWORDS ? 

I would rather this to be in addition because the ALLOWLIST_KEYWORDS are like safelist. I think this behavior is more relevant earlier because there is no safe list so this argument will be a way to define safelist. What do you think?

Reply:
It does and I agree that's probably the most sane thing to do. My thought was to preserve the old behavior if a regex is provided in case someone is using it in unexpected ways, i.e. they only want to compress one particular keyword or something. I think it keeps the script consistent with the old behavior while ensuring the default behavior doesn't cause harm.

I'm actually skeptical this regex feature is used at all and maybe we should consider deprecating it, especially with having a complete list of compressible keywords now. At the very least it is not a usable option when running from ERT.

Reply:
I would also be surprised if anyone is actually using the regex feature.